crates/control/src/game_controller_filter.rs

```diff
@@ -71,9 +89,47 @@ impl GameControllerFilter {
                 hulks_team_is_home_after_coin_toss: game_controller_state_message
                     .hulks_team_is_home_after_coin_toss,
             });
+
+            self.last_contact.insert(*source_address, time);
+            let on_cooldown = self
+                .last_collision_warning
+                .is_some_and(|last_collision_warning| {
+                    time.duration_since(last_collision_warning)
+                        .expect("time ran backwards")
+                        .as_secs_f32()
+                        < 10.0
+                });
+            let recent_contacts = self.last_contact.iter().filter(|(_address, last_contact)| {
+                time.duration_since(**last_contact)
+                    .expect("time ran backwards")
+                    .as_secs_f32()
+                    < 5.0
+            });
+            let collisions = recent_contacts.count() > 1;
```

Comment:
Naming: Add a verb. Maybe "at least two game controllers collided" or something else. Gippedy to your success 💪

Reply:
I renamed this variable.